src/gui/MainWindow.cpp

```diff
@@ -644,17 +649,31 @@ void MainWindow::resetWindowTitle()
 bool MainWindow::mayChangeProject(bool stopPlayback)
 {
 	if( stopPlayback )
+	{
 		Engine::getSong()->stop();
+	}
 
-	if( !Engine::getSong()->isModified() )
+	if( !Engine::getSong()->isModified() && !isRecoverSession() )
 	{
 		return( true );
 	}
 
-	QMessageBox mb( tr( "Project not saved" ),
-				tr( "The current project was modified since "
+	// Separate message strings for modified and recovered files
+	QString messageTitleRecovered = tr( "Recovered project not saved" );
+	QString messageRecovered = tr( "This project was recovered from the "
+					"previous session. It is currently "
+					"unsaved and will be lost if you don't "
+					"Save it. Do you want to save it now?" );
+
+	QString messageTitleUnsaved = tr( "Project not saved" );
```

Comment:
What happens with the already translated string below when I move it to a variable? I didn't change the text. Do I need to do anything to link it to the old string?